prow/config/jobs.go

```diff
@@ -91,7 +91,8 @@ type JobBase struct {
 	// Cluster is the alias of the cluster to run this job in.
 	// (Default: kube.DefaultClusterAlias)
 	Cluster string `json:"cluster,omitempty"`
-	// Namespace is the namespace in which pods schedule.
+	// Namespace is the namespace in which pods schedule. It must only
+	// be set when Agent is "knative"
```

Comment:
Is this true? Don't we honor it for Kubernetes agent?

Reply:
```
$ k logs plank-76cb944746-9pl8d
{"component":"plank","error":"invalid periodic job periodic-kubermatic-stale: namespace customization requires agent: knative-build (found \"kubernetes\")","level":"fatal","msg":"Error starting config agent.","time":"2018-12-08T21:18:40Z"}
```

It would be cool thought to also honor this for jobs that use the Kubernetes agent :)

Reply:
Very interesting. @fejta we should probably bring those two implementations into sync?

Reply:
The resulting question would then be if the `Namespace` here is the `ProwJob` namespace or the pod namespace

Reply:
I'd have to read the code, but is this where the knative Build CRD is created or where the Pod for the Prowjob is created in that impl? @fejta